---
title: "Macrophage Gene Signatures"
author: "Giuliano Netto Flores Cruz"
date: "October 9, 2018"
bibliography: library.bib
csl: abnt-ufrgs-initials.csl
output: word_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

#Defining Macrophage Gene Signatures

  Macrophages have been demonstrated to develop highly complex activation profiles [@Ginhoux2016]. Although many genetic markers are known to play important roles within the macrophage biology context, defining a robust set of gene signatures for the currently known phenotypic subsets remains a challenging task [@Martinez2014]. Recent cytometric and genomic approaches have revealed major limitations regarding the classic M1 versus M2-polarization model, which is no longer accepted as suitable to explain the biologic dynamics of macrophage response [@Martinez2014].
  
  In the pursuit of standardization towards reproducible research, back in 2014 a group of specialists launched nomenclatures and experimental guidelines for the macrophage activation profiles well-established by then [@Murray2014]. However, much of the recent genomic data has challenged the phenotypic techniques used to classify macrophage subsets and therefore novel approaches emerged. In the same year, a multi-center work demonstrated the actual heterogeneity os macrophages through machine learning algorithms applied to single-cell transcriptome analysis [@Xue2014]. 
  
  Assessing the transcriptomes from almost 300 diversely-stimulated human macrophages, Xue and colleagues used weighted gene coexpression network analysis (WGCNA) to identify 49 coexpression modules, which ranged from less than 30 to over 800 different genes [@Xue2014]. Based on Pearson correlation, WGCNA defines gene sets, known as _transcriptional modules_, which present with specific expression patterns across each treatment condition [@Langfelder2008]. These modules could then be used to visualize the comprehensiveness of the M1 versus M2 model. As noted by the authors, stimuli not M1- or M2-associated showed prominent patterns consistent with a rather dynamic spectrum model of cell activation.
  
  In order to achieve high sensivity for potential macrophage phenotypes, the 49 transcriptional modules produced by Xue and colleagues were used as gene sets for further analysis in this study. The Figure 1 shows the distribution of number of genes across the different modules.
  
```{r fig.height = 6, fig.width = 9, fig.align = "center", echo=FALSE, warning=FALSE, out }

library(stringr)
library(ggplot2)
dat <- read.table('/home/giulianonetto/windows/tcc/checklist/genesigs/papers/Xue, 2014/modules.csv', sep = '\t', header = T, stringsAsFactors = F)

dat_colnames <- colnames(dat)
dat_colnames_new = vector()
for (i in 1:ncol(dat)){
  old <- str_extract(dat_colnames[i], "[0-9]+")
  new = as.character(old)
  dat_colnames_new = c(dat_colnames_new, new)
}
colnames(dat) <- dat_colnames_new

# make bar plot for numb of genes per module
colcount <- data.frame()
for(i in 1:ncol(dat)){
  module = colnames(dat)[i]
  Counts =  length(unique(dat[[i]])) - 1
  colcount = rbind(colcount,data.frame(module, Counts))
}

ggplot(colcount, aes(x=module, y=Counts))+
  geom_bar(width = 0.7, position = position_dodge(width = 0.9), stat = "identity")+
  theme(axis.text.x = element_text(angle = 45, hjust = 1.2, face = "bold", size = 11),
        axis.text.y = element_text(face = "bold", size = 15),
        plot.title = element_text(hjust = 0.5, face = "bold", size = 25),
        axis.title.x = element_text(face = "bold", size = 20),
        axis.title.y = element_text(face = "bold", size = 20))+
  ylab("Number of genes\n")+
  xlab("\nModules")+
  ggtitle("Number of genes per module\n")

```
